Phase-5/SearchEngine/Classes/SearchEngineCore.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using SearchEngine.Interfaces;
+
+namespace SearchEngine.Classes
+{
+    public class SearchEngineCore : ISearchEngineCore
+    {
+        private const string SearchWordRegex = "([,.;'\"?!@#$%^&:*]*)([+-]?\\w+)([,.;'\"?!@#$%^&:*]*)";
+        private readonly IInvertedIndex _invertedIndex;
+        public SearchEngineCore(IInvertedIndex invertedIndex) {
+            this._invertedIndex = invertedIndex;
+        }
+        public HashSet<int> search(string statement)
+        {
+            SearchFields fields = new SearchFields();
+
+            Regex rx = new Regex(SearchWordRegex,
+                RegexOptions.Compiled);
+            MatchCollection matches = rx.Matches(statement);
+            foreach (Match match in matches) {
+                GroupCollection groups = match.Groups;
+                string word = groups[2].Value;
+                if (word.StartsWith("+"))
```

Comment:
https://stackoverflow.com/questions/2125066/is-it-a-bad-practice-to-use-an-if-statement-without-curly-braces